build/configs/common_download.sh

```diff
@@ -356,7 +356,7 @@ erase()
 				echo "=========================="
 			fi
 
-			board_erase $TTYDEV ${offsets[$partidx]} ${sizes[partidx]} ${parts[$partidx]}
+			board_erase $TTYDEV ${offsets[$partidx]} ${sizes[partidx]} ${parts[$partidx]} $2
```

Comment:
Hello @jiangxiaoyunjoy 
As I heard that $2 is necessary to distinguish between "ALL" and specific partition.
So if user enter "Erase ALL" command, then download tool erase all partition except the first 512KB.

But when I test "Erase ALL", it seems to erase all partition including first 512KB.
```
Starting Erase...

==========================
      Erasing All
==========================

==========================
Erasing bl1 partition
==========================

==========================
Erasing reserved partition
==========================

==========================
Erasing ftl partition
==========================

==========================
Erasing kernel partition
==========================

==========================
Erasing common partition
==========================

==========================
Erasing app1 partition
==========================

==========================
Erasing kernel partition
==========================

==========================
Erasing common partition
==========================

==========================
Erasing app1 partition
==========================

==========================
Erasing bootparam partition
==========================

==========================
Erasing ext_data partition
==========================

==========================
Erasing resource partition
==========================

==========================
Erasing resource partition
==========================

==========================
Erasing userfs partition
==========================

==========================
Erasing reserved partition
==========================

===================================
Start USB full erase in Flash
===================================
Number of images: 15
Found ACM device         ......    OK
Open serial port         ......    OK

USB erase...
Close serial port        ......    OK
Open serial port         ......    OK
Close serial port        ......    OK
Open serial port         ......    OK
USB Prepare              ......    OK
Flash size 16 MB
Check flash size         ......    OK
Min address to check flash 0x8000000
Flash protection         ......    OK
======================================
erase from 0x8000000, 60KB
======================================
NOR erase 0x8000000 - 0x800f000, 60KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x800f000, 40KB
======================================
NOR erase 0x800f000 - 0x8019000, 40KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x8019000, 12KB
======================================
NOR erase 0x8019000 - 0x801c000, 12KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x8080000, 1844KB
======================================
NOR erase 0x8080000 - 0x824d000, 1844KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x824d000, 5404KB
======================================
NOR erase 0x824d000 - 0x8794000, 5404KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x8794000, 684KB
======================================
NOR erase 0x8794000 - 0x883f000, 684KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x883f000, 1844KB
======================================
NOR erase 0x883f000 - 0x8a0c000, 1844KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x8a0c000, 5404KB
======================================
NOR erase 0x8a0c000 - 0x8f53000, 5404KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x8f53000, 684KB
======================================
NOR erase 0x8f53000 - 0x8ffe000, 684KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x8ffe000, 8KB
======================================
NOR erase 0x8ffe000 - 0x9000000, 8KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x9000000, 128KB
======================================
NOR erase 0x9000000 - 0x9020000, 128KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x9020000, 14656KB
======================================
NOR erase 0x9020000 - 0x9e70000, 14656KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0x9e70000, 14656KB
======================================
NOR erase 0x9e70000 - 0xacc0000, 14656KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0xacc0000, 2048KB
======================================
NOR erase 0xacc0000 - 0xaec0000, 2048KB
USB erase                ......    OK
Partition erase          ......    OK
======================================
erase from 0xaec0000, 1280KB
======================================
NOR erase 0xaec0000 - 0xb000000, 1280KB
USB erase                ......    OK
Partition erase          ......    OK
Close serial port        ......    OK
Complete USB erase

Erase COMPLETE!
```

Reply:
Also, there's some meaningless logs above...

Reply:
Hi @jylee9613 ,

From my understanding, when full erase-> only keep SS unerased, and erase the rest partition? this has been done in common_download.sh erase() function.
If you'd like to add a new request to protect 1st 512KB in "erase all" cmd when written protect enabled, please let me know. I'll update the download tool.

Redundant log in common_download.sh in partition download and erase will be removed after settling the logic of the download scripts. (as @Tirtha-Samsung  also working on it).

Thank you~

Reply:
Thank you for your explaination. You don't need to protect 1st 512KB in "Erase ALL", because I only worried about erasing SS while running erase function.